mm/kfence/core.c

```diff
@@ -348,7 +348,11 @@ static void check_cache_freelist_ptr(int index)
 
 	if (!meta->cache)
 		return;
+#if defined(CONFIG_SLUB)
```

Comment:
We should avoid #ifdef 

```
if (IS_ENABLED(CONFIG_SLAB))
  return;
```

Our Kconfig should take care of ensuring we don't have something other than SL[AU]B, right?

Otherwise we might need a

```
BUILD_BUG_ON(IS_ENABLED(CONFIG_SLUB) || IS_ENABLED(CONFIG_SLAB));
```

Reply:
> We should avoid #ifdef

In this case we cannot. SLAB has no meta->cache->offset, so the code won't compile.
OTOH I think it's cleaner to just #ifdef the whole function body (see the updated patch).

Reply:
> Our Kconfig should take care of ensuring we don't have something other than SL[AU]B, right?

Added this, thanks!